include/psa/crypto_struct.h

```diff
@@ -223,6 +223,29 @@ static inline struct psa_key_derivation_s psa_key_derivation_operation_init(
     return v;
 }
 
+struct psa_key_generation_method_s {
+    /* Future versions may add other fields in this structure. */
+    uint32_t flags;
```

Comment:
Why have `flags` at all? Currently it's unused, but it's a convenient way of packing some information, which can include directives on the content of `data` as well as the validity of yet-to-be-added fields. I expect some of those flags to be vendor-specific.

Reply:
>  Currently it's unused

I think it would be better to start using it actually for RSA. I know that this is not the case right now, but IMO in the client/server scenario that we are aiming to develop in the near/middle future, it might happen that client and server are not exactly aligned to the same code version (for what it concerns with the content of `data`). This field can be split into bit fields to indicate that:
- `data` contains info about RSA key (or another type of key in the future)
- which version format is used to store data in `data`. I mean now this would be `1` (like 1st version, or similar) and it means that `data` only contains the exponent, but in the future there might be follow ups
- [optional] we can keep the last 16bits to indicate also the length of the following data. That would simplify the number of parameters passed to the functions even further.

This is just an idea, not a real request for change :)

Reply:
I would prefer to avoid parsing as much as possible. (Traumatic memories of maintaining a PKCS#11 server.) So `flags` is present to allow some simple extensivity at very low cost, but the current use case does not require any parsing (other than validating `flags==0` for future robustness).

If we want to add more non-boolean inputs to RSA generation in the future, then we can at that time add a flag that says that `data` is interpreted differently. But we'll only have to pay for this complexity if we need it, and I think that's unlikely.